internal/applier/helm.go

```diff
@@ -56,6 +57,25 @@ type Helm struct {
 	Preflights         []Preflight
 }
 
+// shouldSkipPreflight is a helper to determine if the preflight check is CRDUpgradeSafety AND
+// if it is set to enforcement None.
+func shouldSkipPreflight(preflight Preflight, ctx context.Context, ext *ocv1alpha1.ClusterExtension, state string) bool {
+	l := log.FromContext(ctx)
+	if ext.Spec.Install.Preflight != nil && ext.Spec.Install.Preflight.CRDUpgradeSafety != nil {
+		if _, ok := preflight.(*crdupgradesafety.Preflight); ok {
```

Comment:
Maybe move this line out to the calling function, probably a more appropriate place for it, and might solve the linter concern around nested blocks.

Reply:
I resolved this by externalizing the conditions somewhat, but still putting all the 'CRDUpgradeSafety' checks in one place.  
My assertion is still that this kind of thing should be exposed via an interface.